core/src/main/kotlin/org/evomaster/core/problem/rest/service/AbstractRestSampler.kt

```diff
@@ -131,6 +165,25 @@ abstract class AbstractRestSampler : HttpWsSampler<RestIndividual>() {
         log.debug("Done initializing {}", AbstractRestSampler::class.simpleName)
     }
 
+    private fun generateHttpWsAuthInfoFromAuthDto(authDto: AuthenticationDto) : HttpWsAuthenticationInfo {
+
+        // headers for the authenticationDto
+        val headers: MutableList<AuthenticationHeader> = mutableListOf()
+
+        authDto.fixedHeaders.forEach loop@{ h ->
+            val name = h.name?.trim()
+            val value = h.value?.trim()
+            if (name == null || value == null) {
+                throw SutProblemException("Invalid header in ${authDto.name}, $name:$value")
+            }
+
+            headers.add(AuthenticationHeader(name, value))
+        }
+
+        return HttpWsAuthenticationInfo(authDto.name, headers, null)
```

Comment:
is this copy&paste from `HttpWsSampler`? should avoid code duplication. Read directly the results of `HttpWsSampler.authentications`

Reply:
Hi Andrea,

Thanks for your comment. I have refactored the method in which AuthenticationDto is converted to HttpWsAuthentication info as a static method and then used it for this purpose too. The name of the method is:  convertAuthenticationDtoToAuthenticationInfo, which is just refactoring from the previous method handleAuthInfo. I did not change any lines, just moved lines converting from AuthenticationDto to HttpWsAuthenticationInfo into this static method.